R/author.R

```diff
@@ -0,0 +1,139 @@
+
+#' * Explicit use of the fields `Author` or `Maintainer`. We recommend switching
+#' to the more modern `Authors@R` field instead, because it offers richer
+#' metadata for various downstream uses. (Note that `Authors@R` is *eventually*
+#' processed to create `Author` and `Maintainer` fields, but only when the
+#' `tar.gz` is built from package source.)
+
+#' * Presence of the fake author placed by [create_package()] and
+#' [use_description()]. This happens when \pkg{usethis} has to create a
+#' DESCRIPTION file and the user hasn't given any author information via the
+#' `fields` argument or the global option `"usethis.description"`. The
+#' placeholder looks something like `First Last <first.last@example.com> [aut,
+#' cre] (YOUR-ORCID-ID)` and `use_author()` offers to remove it in interactive
+#' sessions.
+#'
+#' @inheritParams utils::person
+#' @export
+#' @examples
+#' \dontrun{
+#' use_author(
+#'   given = "Lucy",
+#'   family = "van Pelt",
+#'   email = "lucy@example.com",
+#'   comment = c(ORCID = "LUCY-ORCID-ID")
+#' )
+#' }
+#'
+use_author <- function(given = NULL, family = NULL, ..., role = "ctb") {
+  check_is_package("use_author()")
+  maybe_name(given)
+  maybe_name(family)
+  check_character(role)
+
+  d <- proj_desc()
+  challenge_legacy_author_fields(d)
+  # We only need to consider Authors@R
+
+  author <- utils::person(given = given, family = family, role = role, ...)
+
+  authors_at_r_already <- d$has_fields("Authors@R")
+  if (authors_at_r_already) {
+    check_author_is_novel(given, family, d)
+  }
+  # This person is not already in Authors@R
+
+  if (authors_at_r_already) {
+    ui_done("
+      Adding to {ui_field('Authors@R')} in DESCRIPTION:
+      {format(author, style = 'text')}")
+  } else {
+    ui_done("
+      Creating {ui_field('Authors@R')} field in DESCRIPTION and adding:
+      {format(author, style = 'text')}")
+  }
+  d$add_author(given = given, family = family, role = role, ...)
+
+  challenge_default_author(d)
+
+  d$write()
+
+  invisible(TRUE)
+
+}
+
+challenge_legacy_author_fields <- function(desc = proj_desc()) {
+  has_legacy_field <-
+    desc$has_fields("Author") || desc$has_fields("Maintainer")
+  if (!has_legacy_field) {
+    return(invisible())
+  }
+
+  ui_oops("
+    Found legacy {ui_field('Author')} and/or {ui_field('Maintainer')} field \\
+    in DESCRIPTION.
+    usethis only supports modification of the {ui_field('Authors@R')} field.")
+  ui_info("
+    We recommend one of these paths forward:
+    * Delete these fields and rebuild with {ui_code('use_author()')}.
+    * Convert to {ui_field('Authors@R')} with {ui_code('desc::desc_coerce_authors_at_r()')},
+      then delete the legacy fields.")
+  if (ui_yeah("Do you want to cancel this operation and sort that out first?")) {
+    ui_stop("Cancelling.")
+  }
+  invisible()
+}
+
+check_author_is_novel <- function(given = NULL, family = NULL, d = proj_desc()) {
```

Comment:
OTOH usethis's "check" functins do really check something and error if the condition does not hold.